streaming/src/main/scala/org/apache/spark/streaming/dstream/SocketInputDStream.scala

```diff
@@ -52,46 +52,60 @@ class SocketReceiver[T: ClassTag](
     storageLevel: StorageLevel
   ) extends Receiver[T](storageLevel) with Logging {
 
+  private var socket: Socket = _
+
   def onStart() {
-    // Start the thread that receives data over a connection
-    new Thread("Socket Receiver") {
-      setDaemon(true)
-      override def run() { receive() }
-    }.start()
+
+    try {
+      logInfo(s"Connecting to $host:$port")
+      socket = new Socket(host, port)
+      logInfo(s"Connected to $host:$port")
+
+      // Start the thread that receives data over a connection
+      new Thread("Socket Receiver") {
+        setDaemon(true)
+        override def run() { receive() }
+      }.start()
+    } catch {
+      case e: ConnectException =>
```

Comment:
Should we give up IOException?  In Original logic, IO Exception(throws from socket = new Socke) also be caughted and trigger the restart logic.